File: server/src/main/java/com/github/genraven/gradlejavaserver/domain/spell/Spell.java
package com.github.genraven.gradlejavaserver.domain.spell;

import com.github.genraven.gradlejavaserver.domain.Difficulty;
import com.github.genraven.gradlejavaserver.domain.skill.Skill;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;

@Data
@Document(collection = "spells")
public class Spell {

    protected Spell() {}

    public Spell(final String name) {
        this.name = name;
    }

    @Id
    private String name;
    private boolean concentration = false;
    private Difficulty difficulty = Difficulty.EASY;
    private String description;
    private List<Skill> skills = new ArrayList<>();
    private List<Effect> effects = new ArrayList<>();
}

File: client/src/models/spell/Spell.ts
import {Difficulty} from "../common/Difficulty";
import Skill from "../actor/Skill";

export default interface Spell {
    id: string
    name: string
    concentration: boolean
    difficulty: Difficulty
    description: string
    skills: Skill[]
    effects: Effect[]
}

export interface Effect {
    name: string
    description: string
    increase: number
}